lib/quicly.c

```diff
@@ -1473,6 +1478,10 @@ static int discard_handshake_context(quicly_conn_t *conn, size_t epoch)
     if ((ret = discard_sentmap_by_epoch(conn, 1u << epoch)) != 0)
         return ret;
     destroy_handshake_flow(conn, epoch);
+    if (epoch == QUICLY_EPOCH_HANDSHAKE) {
+        assert(conn->stash.now != 0);
+        conn->super.stats.handshake_msec = conn->stash.now - conn->created_at;
```

Comment:
Maybe change `handshake_msec` to `handshake_confirmed_msec`?

Yes, it might be important to calculate and collect duration observed at this moment, if our intent is to monitor QUIC stack behavior.

But for other purposes (e.g., HTTP performance monitoring), it makes more sense to call out handshake as done when the application gains the opportunity to send 1-RTT packets.

Considering that, I think we'd better be clear that this value represents a specific moment within the QUIC stack (i.e., "handshake confirmed") rather than a vague concept of "how long the handshake took." Hence the proposed name change.

See also: https://datatracker.ietf.org/doc/html/rfc9001#section-4.1.2